splink/athena/athena_linker.py

```diff
@@ -337,8 +326,10 @@ def _execute_sql_against_backend(self, sql, templated_name, physical_name):
         output_obj = self._table_to_splink_dataframe(templated_name, physical_name)
         return output_obj
 
-    def register_table(self, input, table_name, overwrite=False):
+    def _do_execute_sql(self, sql, templated_name, physical_name):
```

Comment:
I also considered making a `@log_sql` decorator that every subclass would have to add to their `_do_execute_sql()`, instead of using the `_log_and_do_execute_sql()` framework in the superclass. But that is one more thing that each subclass would need to import and remember. I thought this was more centralized, even if it is a bit more indirection for the code reader to understand how `_log_and_do_execute_sql` delegates to `_do_execute_sql`

IDK what do you think?

Reply:
From conversations with Robin, we'd rather stay away from decorators as they can make the code more difficult to follow.

The method you've implemented is more in line with the code used in the wider code base.